packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -0,0 +1,207 @@
+ * @param persistedCache - Cache to store the fetched snapshot.
+ * @param logger - Logger to have telemetry events.
+ * @param hostSnapshotFetchOptions - Options to fetch the snapshot if any. Otherwise default will be used.
+ * @returns - True if the snapshot is cached, false otherwise.
+ */
+export async function prefetchLatestSnapshot(
+    resolvedUrl: IResolvedUrl,
+    getStorageToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+    persistedCache: IPersistedCache = new LocalPersistentCache(),
+    logger: ITelemetryBaseLogger,
+    hostSnapshotFetchOptions: ISnapshotOptions | undefined,
+): Promise<boolean> {
+    const odspLogger = createOdspLogger(ChildLogger.create(logger, "PrefetchSnapshot"));
+    const odspResolvedUrl = getOdspResolvedUrl(resolvedUrl);
+    const snapshotUrl = odspResolvedUrl.endpoints.snapshotStorageUrl;
+    let snapshotOptions: ISnapshotOptions = {
+        deltas: 1,
+        channels: 1,
+        blobs: 2,
+    };
+
+    if (hostSnapshotFetchOptions) {
+        snapshotOptions = {...snapshotOptions, ...hostSnapshotFetchOptions};
+    }
+
+    const storageTokenFetcher = toInstrumentedOdspTokenFetcher(
+        odspLogger,
+        odspResolvedUrl,
+        getStorageToken,
+        true /* throwOnNullToken */,
+    );
+
+    const snapshotCacheEntry: ICacheEntry = {
+        type: snapshotKey,
+        key: "",
+        file: {
+            resolvedUrl: odspResolvedUrl,
+            docId: odspResolvedUrl.hashedDocumentId,
+        },
+    };
+
+    return PerformanceEvent.timedExecAsync(
+        odspLogger,
+        { eventName: "PrefetchLatestSnapshot" },
+        async (event: PerformanceEvent) => {
+            let attempts = 1;
+            const success = await getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
+                if (tokenFetchOptions.refresh) {
+                    attempts = 2;
+                }
+                const storageToken = await storageTokenFetcher(tokenFetchOptions, "PrefetchLatestSnapshot");
+                assert(storageToken !== null, "Storage token should not be null");
+                return fetchSnapshot(
+                    snapshotUrl,
+                    odspResolvedUrl,
+                    snapshotOptions,
+                    storageToken,
+                    odspLogger,
+                    persistedCache,
+                    snapshotCacheEntry,
+                );
+            });
+        event.end({ attempts });
+        return success;
+    }).catch((error) => false);
+}
+
+async function fetchSnapshot(
+    snapshotUrl: string,
+    odspResolvedUrl: IOdspResolvedUrl,
+    snapshotOptions: ISnapshotOptions,
+    storageToken: string,
+    logger: TelemetryLogger,
+    persistedCache: IPersistedCache,
+    snapshotCacheEntry: ICacheEntry,
+): Promise<boolean> {
+    let success = false;
+    const url = `${snapshotUrl}/trees/latest?ump=1`;
+    const formBoundary = uuid();
+    let postBody = `--${formBoundary}\r\n`;
```

Comment:
As a thought: would it be easier if `postBody` was an array of strings, that you would do an `array.join('\r\n')` on at the end?
I think it would look cleaner, and you wouldn't risk forgetting the `\r\n` every time or accidentally overriding due to `=` instead of `+=`

Reply:
Done.